provider/apply.go

```diff
@@ -270,6 +273,46 @@ func (s *RawProviderServer) ApplyResourceChange(ctx context.Context, req *tfprot
 			return resp, nil
 		}
 
+		// wait for the resource to be deleted
+		s.logger.Trace("[ApplyResourceChange][Apply]", "Waiting for resource to be deleted...")
+		timeout := defaultDeleteTimeout
+		if v, ok := priorStateVal["timeouts"]; ok && !v.IsNull() {
+			timeouts := map[string]tftypes.Value{}
+			v.As(&timeouts)
+			if v, ok := timeouts["delete"]; ok && !v.IsNull() {
+				v.As(&timeout)
+			}
+		}
+		s.logger.Trace("[ApplyResourceChange][Apply]", "Using timeout %q for delete", timeout)
+		duration, err := time.ParseDuration(timeout)
+		if err != nil {
+			resp.Diagnostics = append(resp.Diagnostics,
+				&tfprotov5.Diagnostic{
+					Severity: tfprotov5.DiagnosticSeverityError,
+					Detail:   err.Error(),
+					Summary:  fmt.Sprintf("error parsing timeout %q: %v", timeout, err),
+				})
+			return resp, nil
+		}
+		end := time.Now().Add(duration)
+		for {
+			_, err := rs.Get(ctx, rname, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				s.logger.Trace("[ApplyResourceChange][Apply]", "Resource is deleted")
+				break
+			}
+			time.Sleep(1 * time.Second)
```

Comment:
Could we maybe make use of a retry function instead of Sleep? Or ignore the lint error:

https://github.com/bflad/tfproviderlint/tree/main/passes/R018

I wonder if it's possible to mimic what we have in the Kubernetes provider:

https://github.com/hashicorp/terraform-provider-kubernetes/blob/6eb06c1423a81ce487ae75a3e79ae0a0f8ec18b2/kubernetes/resource_kubernetes_deployment.go#L378

Making our own retry function. We could reuse it for creates, deletes, and updates.

https://github.com/hashicorp/terraform-provider-kubernetes/blob/6eb06c1423a81ce487ae75a3e79ae0a0f8ec18b2/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource/wait.go#L15

Or maybe we could make use of a simple context to do the timeout. That is, specifying the timeout inside the context and passing that context to the delete call.

https://golang.org/pkg/context/#WithTimeout

Reply:
> Could we maybe make use of a retry function instead of Sleep? Or ignore the lint error:
>
>https://github.com/bflad/tfproviderlint/tree/main/passes/R018
>
>I wonder if it's possible to mimic what we have in the Kubernetes provider:
>
>https://github.com/hashicorp/terraform-provider-kubernetes/blob/6eb06c1423a81ce487ae75a3e79ae0a0f8ec18b2/kubernetes/resource_kubernetes_deployment.go#L378
>
> Making our own retry function. We could reuse it for creates, deletes, and updates.

I agree having the retry function or StateChangeConf from the SDK would be nice here but to reimplement here we will have to use a sleep anyway and silence the lint anyway. For the moment, I'm going to ignore the lint error and we can make a more re-usable retry in another PR. 

> Or maybe we could make use of a simple context to do the timeout. That is, specifying the timeout inside the context and passing that context to the delete call.
>
>https://golang.org/pkg/context/#WithTimeout

This wont work because the Delete call doesn't actually block, so we have to do the waiting ourselves.